### PR TITLE
kitsas: init at 2.3

### PR DESCRIPTION
**Pull request by gspia**
kitsas: init at 2.3

###### Motivation for this change

This adds an accounting application called kitsas to the set of available applications. Kitsas is suitable for Finnish associations and small business.

This is my first packaged application and related pull request. (I have no exp on reviewing nixpkgs issues yet.) 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115270` at 5cb46d27 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/3bcec4aa4a509daeb67a83d56d80d317">kitsas</a></li></li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/office/kitsas/default.nix:20:3:
```
   |
20 |   configurePhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/office/kitsas/default.nix:29:3:
```
   |
29 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/office/kitsas/default.nix:43:5:
```
   |
43 |     license = lib.licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `qtcreator`.
Near pkgs/applications/office/kitsas/default.nix:1:53:
```
  |
1 | { lib, mkDerivation, fetchFromGitHub, qmake, qtsvg, qtcreator, libsForQt5, libzip, pkg-config  }:
  |                                                     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/office/kitsas/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    cp $src/kitsas.desktop $out/share
+    chmod 644 $out/share/kitsas.desktop
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/artoh/kitupiikki";
+    description = "An accounting tool suitable for Finnish assocications and small business";
+    maintainers = [ "gspia" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:42]:
```suggestion
    maintainers = with maintainers; [ gspia ];
```
Also please add yourself to the maintainers list in a separate commit.

File: pkgs/applications/office/kitsas/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    license = lib.licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:43]:
```suggestion
    license = licenses.gpl3;
```

File: pkgs/applications/office/kitsas/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  configurePhase = ''
+    mkdir build-linux
+    cd build-linux
+    qmake $src/kitsas/kitsas.pro -spec linux-g++ "CONFIG+=release" && make qmake_all
+  '';
+  preFixup = ''
+    make clean
+    rm Makefile
+  '';
+  installPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:29]:
```suggestion
  '';

  installPhase = ''
```

File: pkgs/applications/office/kitsas/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  phases = [ "unpackPhase" "configurePhase" "buildPhase" "installPhase" "fixupPhase" ];
+
+  # We use a separate build-dir as otherwise ld seems to get confused between
+  # directory and executable name on buildPhase.
+  configurePhase = ''
+    mkdir build-linux
+    cd build-linux
+    qmake $src/kitsas/kitsas.pro -spec linux-g++ "CONFIG+=release" && make qmake_all
+  '';
+  preFixup = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:25]:
```suggestion
  '';

  preFixup = ''
```

File: pkgs/applications/office/kitsas/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    sha256 = "1qac6cxkb45rs5pschsf2rvpa789g27shmrwpshwahqzhw42xvgl";
+  };
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ qmake qtsvg libsForQt5.poppler libzip ];
+
+  phases = [ "unpackPhase" "configurePhase" "buildPhase" "installPhase" "fixupPhase" ];
+
+  # We use a separate build-dir as otherwise ld seems to get confused between
+  # directory and executable name on buildPhase.
+  configurePhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:20]: Can you convert this into preConfigure and qmakeFlags?

File: pkgs/applications/office/kitsas/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  src = fetchFromGitHub {
+    owner = "artoh";
+    repo = "kitupiikki";
+    rev = "v" + version;
+    sha256 = "1qac6cxkb45rs5pschsf2rvpa789g27shmrwpshwahqzhw42xvgl";
+  };
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ qmake qtsvg libsForQt5.poppler libzip ];
+
+  phases = [ "unpackPhase" "configurePhase" "buildPhase" "installPhase" "fixupPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:16]:
```suggestion
```

File: pkgs/applications/office/kitsas/default.nix [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  pname = "kitsas";
+  version = "2.3";
+
+  src = fetchFromGitHub {
+    owner = "artoh";
+    repo = "kitupiikki";
+    rev = "v" + version;
+    sha256 = "1qac6cxkb45rs5pschsf2rvpa789g27shmrwpshwahqzhw42xvgl";
+  };
+  nativeBuildInputs = [ pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:13]:
```suggestion
  };

  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/applications/office/kitsas/default.nix [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  buildInputs = [ qmake qtsvg libsForQt5.poppler libzip ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:14]:
```suggestion
  buildInputs = [ qmake qtsvg poppler libzip ];
```

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -23393,6 +23393,8 @@ in
     ffmpeg = ffmpeg_2;
   };
 
+  kitsas = qt5.callPackage ../applications/office/kitsas { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:23396]:
```suggestion
  kitsas = libsForQt5.callPackage ../applications/office/kitsas { };
```

File: pkgs/applications/office/kitsas/default.nix [hunk 10]
```diff
@@ -0,0 +1,47 @@
+{ lib, mkDerivation, fetchFromGitHub, qmake, qtsvg, qtcreator, libsForQt5, libzip, pkg-config  }:
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:1]:
```suggestion
{ lib, mkDerivation, fetchFromGitHub, qmake, qtsvg, qtcreator, poppler, libzip, pkg-config  }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/office/kitsas/default.nix (hunk 1 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Added gspia into maintainers in one commit and after it made the mainainers-line change in another commit.

File: pkgs/applications/office/kitsas/default.nix (hunk 2 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok, rewritten now

File: pkgs/applications/office/kitsas/default.nix (hunk 3 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok added newline

File: pkgs/applications/office/kitsas/default.nix (hunk 4 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok added newline

File: pkgs/applications/office/kitsas/default.nix (hunk 7 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok added newline

File: pkgs/applications/office/kitsas/default.nix (hunk 5 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: This was a bit more difficult change for me, but I think I got it made now. Thanks for suggesting this one (I think some new "clicks" happened in my head now - small but anyway).

File: pkgs/applications/office/kitsas/default.nix (hunk 6 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok, removed now.

File: pkgs/applications/office/kitsas/default.nix (hunk 8 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok, made this change as well.

File: pkgs/top-level/all-packages.nix (hunk 9 above)
Comment (gspia) [reply pkgs/top-level/all-packages.nix]: And this one too.

File: pkgs/applications/office/kitsas/default.nix (hunk 10 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Removed libsForQt5 and added poppler.

**Comment by gspia** [comment]
Hopefully the commits do what you asked @SuperSandro2000 . Could you please re-review the changes?

File: pkgs/applications/office/kitsas/default.nix [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/artoh/kitupiikki";
+    description = "An accounting tool suitable for Finnish assocications and small business";
+    maintainers = with maintainers; [ gspia ];
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:51]: Final new line to much.

File: pkgs/applications/office/kitsas/default.nix [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/share
+    cp kitsas $out/bin
+    cp $src/kitsas.png $out/share
+    cp $src/kitsas.desktop $out/share
+  '';
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:41]:
```suggestion
```

set by qmake.

File: pkgs/applications/office/kitsas/default.nix [hunk 13]
```diff
@@ -0,0 +1,51 @@
+{ lib, mkDerivation, fetchFromGitHub, qmake, qtsvg, qtcreator, poppler, libzip, pkg-config }:
+
+mkDerivation rec {
+  pname = "kitsas";
+  version = "2.3";
+
+  src = fetchFromGitHub {
+    owner = "artoh";
+    repo = "kitupiikki";
+    rev = "v" + version;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/office/kitsas/default.nix (hunk 11 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: ok, removed now

File: pkgs/applications/office/kitsas/default.nix (hunk 12 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok removed this line now.

File: pkgs/applications/office/kitsas/default.nix (hunk 13 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: and also changed this

**Comment by gspia** [comment]
Thanks for the snd round @SuperSandro2000 ! How this MR looks now?

While implementing the changes, I started to wonder if there is a tool available to check the style. And possibly some of the facts like the one that using qmake means enabling parallel building. E.g. something like `hlint` that is used in Haskell. (I'll probably start a thread at nixos discourse as if there are tools, I think other people would benefit knowing about those - I tried to search for some but didn't find anything yet.)

**Comment by rmcgibbo** [comment]
There are a few formatters (I use `nixpkgs-format`), and there's also `nixpkgs-hammering` to give hints. Can you squash the commits down into 2? It should be two commits like "maintainers: add {name}", and then "kitsas: init at 2.3".

**Comment by gspia** [comment]
Haa!
I hope this has now better shape and sorry for not realizing right in the beginning to squash also the review changes into two commits and using the correctly formatted commit titles. 
Thus, could you please check once more @rmcgibbo and/or @SuperSandro2000 that everything is ok?

File: pkgs/applications/office/kitsas/default.nix [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    cp kitsas $out/bin
+    cp $src/kitsas.png $out/share
+    cp $src/kitsas.desktop $out/share
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/artoh/kitupiikki";
+    description = "An accounting tool suitable for Finnish assocications and small business";
+    maintainers = with maintainers; [ gspia ];
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:45]: gpl3Omly or gpl3Plus?

Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix:45]: I think it is gpl3 only - but I'm not a lawer and probably won't recognize easily the difference between licenses.
The nix repos seem to have over 2600 gpl3, about 100 gpl only's and a bit over 800 gpl3 plus licensed packages.
Is this important distinction?

Comment (SuperSandro2000) [reply pkgs/applications/office/kitsas/default.nix:45]: gpl3 is deprecated and eventually all those packages should be converted :) so we try to not add any new anymore.

According to their license text you can use any later version https://github.com/artoh/kitupiikki/blob/master/LICENSE#L7 so it is gpl3Plus.

Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Interesting and thanks! Now while you show it, this is quite clear. I tried to search (ddg, nixos discourse) about the lpg3-licenses but had difficulties finding anything relevant. (Maybe I used from search phrase. And now I don't remember if I looked at the nixpkgs-manual.)

Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: I'll resolve this gpl3 thread now and as for re-review again

File: pkgs/applications/office/kitsas/default.nix [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  preFixup = ''
+    make clean
+    rm Makefile
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/share
+    cp kitsas $out/bin
+    cp $src/kitsas.png $out/share
+    cp $src/kitsas.desktop $out/share
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:38]:
```suggestion
    cp $src/kitsas.desktop $out/share/applications
```

File: pkgs/applications/office/kitsas/default.nix [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+  preFixup = ''
+    make clean
+    rm Makefile
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/share
+    cp kitsas $out/bin
+    cp $src/kitsas.png $out/share
```
Comment (SuperSandro2000) [inline pkgs/applications/office/kitsas/default.nix:37]: This probably needs adjustment, too.

File: pkgs/applications/office/kitsas/default.nix (hunk 15 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Thanks, I made this change and tried two formatters on this `default.nix` file and included the changes (the `qmakeFlags` on separate lines). Further, I found a typo on the description (the associations word).

To be sure that there is that `applications` directory, I added it into the `mkdir` command above.

I didn't realize at all that the `applications` directory was needed (the icons were "working" in my case when I tried this locally).

File: pkgs/applications/office/kitsas/default.nix (hunk 16 above)
Comment (gspia) [reply pkgs/applications/office/kitsas/default.nix]: Ok, I copied this too into the `applications` directory.

**Pull request merged**